packages/babel-plugin-proposal-class-properties/test/fixtures/private-loose/optional-chain-cast-to-boolean/output.js

```diff
@@ -0,0 +1,127 @@
+var _a = babelHelpers.classPrivateFieldLooseKey("a");
+
+var C = /*#__PURE__*/function () {
+  "use strict";
+
+  function C() {
+    babelHelpers.classCallCheck(this, C);
+  }
+
+  babelHelpers.createClass(C, null, [{
+    key: "testIf",
+    value: function testIf(o) {
+      if (o !== null && o !== void 0 && babelHelpers.classPrivateFieldLooseBase(o, _a)[_a].b.c.d) {
+        return true;
+      }
+
+      return false;
+    }
+  }, {
+    key: "testConditional",
+    value: function testConditional(o) {
+      return (o === null || o === void 0 ? void 0 : babelHelpers.classPrivateFieldLooseBase(o, _a)[_a].b)?.c.d ? true : false;
```

Comment:
The output here is not ideal because `willPathCastToBoolean` stops at the optional chain `?.c.d`. I think such cases are rare and we could iterate later.